frontend/src/explorer/ExplorerWebSocketContainer.tsx

```diff
@@ -36,7 +42,36 @@ export const ExplorerWebSocketContainer = ({
   readOnly,
 }: ExplorerWebSocketContainerProps) => {
   const [state, dispatch] = useReducer(reducer, initialState);
-  const { viewState, id, tree, expanded, maxDepth, message } = state;
+  const { viewState, id, tree, expanded, maxDepth, synchronized, message } = state;
+
+  const [getTreePath, { loading: treePathLoading, data: treePathData, error: treePathError }] = useLazyQuery<
+    GQLGetTreePathData,
+    GQLGetTreePathVariables
+  >(getTreePathQuery);
+
+  useEffect(() => {
+    if (tree) {
+      const variables: GQLGetTreePathVariables = {
+        editingContextId,
+        treeId: tree.id,
+        selectionEntryIds: synchronized ? selection.entries.map((entry) => entry.kind + '&id=' + entry.id) : [],
```

Comment:
This should be an id, not a custom made data. We can't trust GraphQL calls to create the proper data format, the frontend should not be trusted to create coherent derived information. The backend should figured out the type of the data from its id. It gave it to the frontend after all.